dask_sql/mappings.py

```diff
@@ -280,6 +292,14 @@ def cast_column_to_type(col: dd.Series, expected_type: str):
     """Cast the given column to the expected type"""
     current_type = col.dtype
 
+    try:
+        from cudf.api.types import is_decimal_dtype
+
+        if is_decimal_dtype(current_type):
+            return col
```

Comment:
Do we run into errors when trying to cast a decimal column to a different type? If so and we need to avoid conversion, then we should probably change this to

```suggestion
            return None
```

To be consistent with the behavior below when conversion is skipped